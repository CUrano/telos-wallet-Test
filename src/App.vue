<template>
  <div id="q-app">
    <router-view />
  </div>
</template>
<script>

import { Component, Vue } from "vue-property-decorator";
import { vxm } from "./store/index.js";
import wait from "waait";

export default {
  name: 'App',
  methods: {
    async loadBancor() {
      const trade = this.$route.meta.feature == "Trade";

      const service = this.$route.params && this.$route.params.service;
      const pool = this.$route.params && this.$route.params.account;
      const feature = this.$route.meta && this.$route.meta.feature;
      const query = this.$route.query;
      const paramsSatisfied = service && feature && query;

      const initParams = {
        initialChain: this.$route.params.service,
        ...(paramsSatisfied && {
          initialModuleParam: {
            [trade ? "tradeQuery" : "poolQuery"]: trade ? this.$route.query : pool
          }
        })
      };

      try {
        await vxm.bancor.init(initParams);
        this.loading = false;
      } catch (e) {
        await wait(1000);
        try {
          await vxm.bancor.init(initParams);
        } catch (e) {
          this.loading = false;
          this.error = e.message;
          throw new Error(e);
        }
      }
    }
  },
  async created() {
    this.loadBancor();
  }
}
</script>
