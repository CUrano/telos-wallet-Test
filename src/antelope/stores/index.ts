import { App } from 'vue';
import { StoreCallback } from '@quasar/app-webpack';
import { store } from 'quasar/wrappers';
import { createPinia } from 'pinia';

export default store(((app: App) => {
    const pinia = createPinia();
    app.use(pinia);
    return pinia;
}) as StoreCallback);
