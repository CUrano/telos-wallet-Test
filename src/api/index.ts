import {
    getCoingeckoUsdPrice,
    getCoingeckoPriceChartData,
    getEmptyPriceChartData,
} from 'src/api/price';

export const api = {
    getCoingeckoUsdPrice,
    getCoingeckoPriceChartData,
    getEmptyPriceChartData,
};

let _useAntelopeLib = localStorage.getItem('useAntelopeLib') === 'true';
export const setUseAntelopeLib = (value: boolean) => {
    _useAntelopeLib = value;
    localStorage.setItem('useAntelopeLib', value ? 'true' : 'false');
};
export const useAntelopeLib = () => _useAntelopeLib;
