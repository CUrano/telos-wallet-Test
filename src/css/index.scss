@import 'legacy-styles';

// components
@import 'components/notification';
@import 'components/inputs';
