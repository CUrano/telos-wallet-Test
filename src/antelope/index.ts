import { App } from 'vue';
import { AntelopeConfig } from 'src/antelope/config/';
import installPinia from 'src/antelope/stores';

import { useUserStore } from 'src/antelope/stores/user';
import { ChainModel, useChainStore } from 'src/antelope/stores/chain';
import { AccountModel, useAccountStore } from 'src/antelope/stores/account';
import { useProfileStore } from 'src/antelope/stores/profile';
import { useResourcesStore } from 'src/antelope/stores/resources';
import { useRexStore } from 'src/antelope/stores/rex';
import { useTokensStore } from 'src/antelope/stores/tokens';
import { useContractStore } from 'src/antelope/stores/contract';
import { useBalancesStore } from 'src/antelope/stores/balances';
import { useHistoryStore } from 'src/antelope/stores/history';
import { useFeedbackStore } from 'src/antelope/stores/feedback';
import { usePlatformStore } from 'src/antelope/stores/platform';
import { useEVMStore } from 'src/antelope/stores/evm';
import { Subject } from 'rxjs';

const events = {
    onLoggedIn: new Subject<AccountModel>(),
    onLoggedOut: new Subject<void>(),
    onChainChanged: new Subject<{label:string, chain:ChainModel}>(),
    onAccountChanged: new Subject<{label:string, account:AccountModel|null}>(),
    onErrorMessage: new Subject<{name: string, message:string}>(),
};

export class Antelope {
    constructor(
        public config: AntelopeConfig,
    ) {
        //
    }

    init(app: App) {
        if (this.config.app) {
            throw new Error('Antelope has already been initialized.');
        }
        this.config.init(app);

        // do not access pinia stores before this line
        installPinia(app);

        // call for the first time useXStore for all X stores in Antelope library
        const stores = this.stores;

        // call init to all stores
        Object.keys(stores).forEach((key) => {
            const store = (stores as unknown as {[k:string]:{init?:()=>void}})[key];
            if (store.init) {
                store.init();
            }
        });

        // Initializing store
        stores.user.loadUsers();
    }

    get stores() {
        return {
            user: useUserStore(),
            chain: useChainStore(),
            account: useAccountStore(),
            profile: useProfileStore(),
            resources: useResourcesStore(),
            rex: useRexStore(),
            tokens: useTokensStore(),
            contract: useContractStore(),
            balances: useBalancesStore(),
            history: useHistoryStore(),
            feedbacl: useFeedbackStore(),
            platform: usePlatformStore(),
            evm: useEVMStore(),
        };
    }

    // events for antelope
    get events() {
        return events;
    }
}

const antelope = new Antelope(new AntelopeConfig());
export const getAntelope = () => antelope;
export const installAntelope = (app: App) => {
    if (app.config.globalProperties.$antelope) {
        console.warn(
            'Antelope has already been installed. Vue.use(Antelope) should be called only once.\n' +
            'If you need to get the instance of Antelope, use getAntelope().');
        return antelope;
    }
    antelope.init(app);
    app.config.globalProperties.$antelope = antelope;
    return antelope;
};
