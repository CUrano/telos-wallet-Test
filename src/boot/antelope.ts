import { boot } from 'quasar/wrappers';
import { installAntelope } from 'src/antelope';

export default boot(({ app }) => {
    const ant = installAntelope(app);

    // settting notifucation handlers --
    ant.config.setNotifyErrorHandler(app.config.globalProperties.$errorNotification);
    ant.config.setNotifySuccessHandler(app.config.globalProperties.$successNotification);
    ant.config.setNotifyWarningHandler(app.config.globalProperties.$unexpectedErrorNotification);

    // setting log in and out callbacks --
    ant.events.onLoggedIn.subscribe({
        next: () => {
            const $router = app.config.globalProperties.$router;
            if ($router.currentRoute.value.path === '/') {
                $router.push({ path: '/balance' });
            }
        },
    });
    ant.events.onLoggedOut.subscribe({
        next: () => {
            const $router = app.config.globalProperties.$router;
            if ($router.currentRoute.value.path !== '/') {
                $router.push({ path: '/' });
            }
        },
    });

    // setting authenticators getter --
    ant.config.setAuthenticatorsGetter(() => app.config.globalProperties.$ual.getAuthenticators().availableAuthenticators);

    // setting translation handler --
    ant.config.setLocalizationHandler((key:string) => app.config.globalProperties.$t(key));
});
