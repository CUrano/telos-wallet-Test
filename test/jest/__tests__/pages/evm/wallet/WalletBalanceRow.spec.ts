import { shallowMount } from '@vue/test-utils';

const fakeBuyMoreLink = 'fake';
const fakeStlosContractAddress = '0x'.concat('9'.repeat(40));
const fakeWtlosContractAddress = '0x'.concat('8'.repeat(40));
const chainStoreMock = {
    useChainStore: () => ({
        currentChain: {
            settings: {
                getBuyMoreOfTokenLink: () => fakeBuyMoreLink,
                getStlosContractAddress: () => fakeStlosContractAddress,
                getWtlosContractAddress: () => fakeWtlosContractAddress,
            },
        },
    }),
};

import WalletBalanceRow from 'pages/evm/wallet/WalletBalanceRow.vue';
import * as antelope from 'src/antelope';
import { stubWithSlot } from 'test/jest/testing-helpers';

jest.mock('src/antelope', () => chainStoreMock);
describe('WalletBalanceRow.vue', () => {
    const stubs = {
        'q-tooltip': stubWithSlot('q-tooltip'),
        'q-btn-dropdown': stubWithSlot('q-btn-dropdown'),
    };

    it('should have the correct name', () => {
        expect(WalletBalanceRow.name).toBe('WalletBalanceRow');
    });

    it('should render correctly on desktop devices', () => {
        const wrapper = shallowMount(WalletBalanceRow, {
            global: {
                stubs,
                mocks: {
                    $q: {
                        screen: {
                            lt: {
                                lg: false,
                            },
                        },
                    },
                },
            },
            props: {
                token: {
                    address: fakeStlosContractAddress,
                    symbol: 'STLOS',
                    name: 'Staked TLOS',
                    logoURI: 'https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png',
                    decimals: 18,
                    balance: '3642.0243',
                    fullBalance: '3642.024318091460206147',
                },
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    it('should render correctly on mobile devices', () => {
        const wrapper = shallowMount(WalletBalanceRow, {
            global: {
                stubs,
                mocks: {
                    $q: {
                        screen: {
                            lt: {
                                lg: true,
                            },
                        },
                    },
                },
            },
            props: {
                token: {
                    address: fakeStlosContractAddress,
                    symbol: 'STLOS',
                    name: 'Staked TLOS',
                    logoURI: 'https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png',
                    decimals: 18,
                    balance: '3642.0243',
                    fullBalance: '3642.024318091460206147',
                },
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    describe('should render the right overflow items when the token is', () => {
        test('TLOS', () => {
            const wrapper = shallowMount(WalletBalanceRow, {
                global: {
                    stubs,
                    mocks: {
                        $q: {
                            screen: {
                                lt: {
                                    lg: true,
                                },
                            },
                        },
                    },
                },
                props: {
                    token: {
                        symbol: 'TLOS',
                        name: 'Telos',
                        logoURI: 'https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png',
                        decimals: 18,
                        balance: '3642.0243',
                        fullBalance: '3642.024318091460206147',
                    },
                    tokenIsTlos: true,
                },
            });

            // snapshot contains c-wallet-balance-row__overflow-li for stake, buy, wrap, send
            expect(wrapper.element).toMatchSnapshot();
        });

        test('STLOS', () => {
            const wrapper = shallowMount(WalletBalanceRow, {
                global: {
                    stubs,
                    mocks: {
                        $q: {
                            screen: {
                                lt: {
                                    lg: true,
                                },
                            },
                        },
                    },
                },
                props: {
                    token: {
                        address: fakeStlosContractAddress,
                        symbol: 'STLOS',
                        name: 'Staked TLOS',
                        logoURI: 'https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png',
                        decimals: 18,
                        balance: '3642.0243',
                        fullBalance: '3642.024318091460206147',
                    },
                },
            });

            // snapshot contains c-wallet-balance-row__overflow-li for stake, contract, send
            expect(wrapper.element).toMatchSnapshot();
        });

        test('WTLOS', () => {
            const wrapper = shallowMount(WalletBalanceRow, {
                global: {
                    stubs,
                    mocks: {
                        $q: {
                            screen: {
                                lt: {
                                    lg: true,
                                },
                            },
                        },
                    },
                },
                props: {
                    token: {
                        address: fakeWtlosContractAddress,
                        symbol: 'WTLOS',
                        name: 'Wrapped TLOS',
                        logoURI: 'https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png',
                        decimals: 18,
                        balance: '3642.0243',
                        fullBalance: '3642.024318091460206147',
                    },
                },
            });

            // snapshot contains c-wallet-balance-row__overflow-li for unwrap, contract, send
            expect(wrapper.element).toMatchSnapshot();
        });

        test('not a system token', () => {
            const wrapper = shallowMount(WalletBalanceRow, {
                global: {
                    stubs,
                    mocks: {
                        $q: {
                            screen: {
                                lt: {
                                    lg: true,
                                },
                            },
                        },
                    },
                },
                props: {
                    token: {
                        address: '0x'.concat('1'.repeat(40)),
                        symbol: 'SHIB',
                        name: 'Shiba',
                        logoURI: 'https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png',
                        decimals: 18,
                        balance: '3642.0243',
                        fullBalance: '3642.024318091460206147',
                    },
                },
            });

            // snapshot contains c-wallet-balance-row__overflow-li for contract, send
            expect(wrapper.element).toMatchSnapshot();
        });
    });
});
