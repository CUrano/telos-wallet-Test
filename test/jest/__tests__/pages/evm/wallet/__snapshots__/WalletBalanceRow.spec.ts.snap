// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletBalanceRow.vue should render correctly on desktop devices 1`] = `
<div
  class="c-wallet-balance-row"
>
  <div
    class="c-wallet-balance-row__left-container"
  >
    <img
      alt="evm_wallet.token_logo_alt"
      aria-hidden="true"
      class="c-wallet-balance-row__logo"
      height="40"
      src="https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png"
      width="40"
    />
    <div>
      <p
        class="q-mb-xs"
      >
        STLOS
      </p>
       Staked TLOS
    </div>
  </div>
  <div
    class="c-wallet-balance-row__right-container"
  >
    <div
      class="c-wallet-balance-row__balance-container"
    >
      <div
        class="c-wallet-balance-row__primary-amount"
      >
        3,642.0243 STLOS 
        <!--v-if-->
      </div>
      <span
        class="c-wallet-balance-row__secondary-amount"
      >
        $456,789,123.67 (@ $0.7732 / STLOS)
      </span>
    </div>
    <q-btn-stub
      aria-label="evm_wallet.balance_row_actions_aria"
      class="c-wallet-balance-row__overflow"
      dense=""
      flat=""
      icon="more_vert"
      no-icon-animation=""
    >
      
      <q-menu-stub
        anchor="bottom end"
        offset="0,16"
        self="top right"
      >
        
        <ul
          class="c-wallet-balance-row__overflow-ul"
        >
          
          <li
            aria-labelledby="overflow-text-0"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon c-wallet-balance-row__overflow-icon--stroke"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-0"
            >
              evm_wallet.unstake
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-1"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-1"
            >
              global.contract
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-2"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-2"
            >
              evm_wallet.send
            </span>
          </li>
          
        </ul>
        
      </q-menu-stub>
      
    </q-btn-stub>
  </div>
</div>
`;

exports[`WalletBalanceRow.vue should render correctly on mobile devices 1`] = `
<div
  class="c-wallet-balance-row"
>
  <div
    class="c-wallet-balance-row__left-container"
  >
    <img
      alt="evm_wallet.token_logo_alt"
      aria-hidden="true"
      class="c-wallet-balance-row__logo"
      height="40"
      src="https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png"
      width="40"
    />
    <div>
      <p
        class="q-mb-xs"
      >
        STLOS
      </p>
       Staked TLOS
    </div>
  </div>
  <div
    class="c-wallet-balance-row__right-container"
  >
    <div
      class="c-wallet-balance-row__balance-container"
    >
      <div
        class="c-wallet-balance-row__primary-amount"
      >
        $456.79M 
        
        <inline-svg-stub
          aria-hidden="true"
          class="c-wallet-balance-row__info-icon"
          keepduringloading="true"
          src=""
          transformsource="[Function]"
        />
        <q-tooltip-stub
          model-value="false"
        >
          
          <span
            class="c-wallet-balance-row__tooltip"
          >
            evm_wallet.fiat_value:
$456,789,123.67

evm_wallet.token_balance:
3,642.024300000000000000
          </span>
          
        </q-tooltip-stub>
        
      </div>
      <span
        class="c-wallet-balance-row__secondary-amount"
      >
        3.64K STLOS
      </span>
    </div>
    <q-btn-stub
      aria-label="evm_wallet.balance_row_actions_aria"
      class="c-wallet-balance-row__overflow"
      dense=""
      flat=""
      icon="more_vert"
      no-icon-animation=""
    >
      
      <q-menu-stub
        anchor="bottom end"
        offset="0,16"
        self="top right"
      >
        
        <ul
          class="c-wallet-balance-row__overflow-ul"
        >
          
          <li
            aria-labelledby="overflow-text-0"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon c-wallet-balance-row__overflow-icon--stroke"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-0"
            >
              evm_wallet.unstake
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-1"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-1"
            >
              global.contract
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-2"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-2"
            >
              evm_wallet.send
            </span>
          </li>
          
        </ul>
        
      </q-menu-stub>
      
    </q-btn-stub>
  </div>
</div>
`;

exports[`WalletBalanceRow.vue should render the right overflow items when the token is STLOS 1`] = `
<div
  class="c-wallet-balance-row"
>
  <div
    class="c-wallet-balance-row__left-container"
  >
    <img
      alt="evm_wallet.token_logo_alt"
      aria-hidden="true"
      class="c-wallet-balance-row__logo"
      height="40"
      src="https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png"
      width="40"
    />
    <div>
      <p
        class="q-mb-xs"
      >
        STLOS
      </p>
       Staked TLOS
    </div>
  </div>
  <div
    class="c-wallet-balance-row__right-container"
  >
    <div
      class="c-wallet-balance-row__balance-container"
    >
      <div
        class="c-wallet-balance-row__primary-amount"
      >
        $456.79M 
        
        <inline-svg-stub
          aria-hidden="true"
          class="c-wallet-balance-row__info-icon"
          keepduringloading="true"
          src=""
          transformsource="[Function]"
        />
        <q-tooltip-stub
          model-value="false"
        >
          
          <span
            class="c-wallet-balance-row__tooltip"
          >
            evm_wallet.fiat_value:
$456,789,123.67

evm_wallet.token_balance:
3,642.024300000000000000
          </span>
          
        </q-tooltip-stub>
        
      </div>
      <span
        class="c-wallet-balance-row__secondary-amount"
      >
        3.64K STLOS
      </span>
    </div>
    <q-btn-stub
      aria-label="evm_wallet.balance_row_actions_aria"
      class="c-wallet-balance-row__overflow"
      dense=""
      flat=""
      icon="more_vert"
      no-icon-animation=""
    >
      
      <q-menu-stub
        anchor="bottom end"
        offset="0,16"
        self="top right"
      >
        
        <ul
          class="c-wallet-balance-row__overflow-ul"
        >
          
          <li
            aria-labelledby="overflow-text-0"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon c-wallet-balance-row__overflow-icon--stroke"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-0"
            >
              evm_wallet.unstake
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-1"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-1"
            >
              global.contract
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-2"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-2"
            >
              evm_wallet.send
            </span>
          </li>
          
        </ul>
        
      </q-menu-stub>
      
    </q-btn-stub>
  </div>
</div>
`;

exports[`WalletBalanceRow.vue should render the right overflow items when the token is TLOS 1`] = `
<div
  class="c-wallet-balance-row"
>
  <div
    class="c-wallet-balance-row__left-container"
  >
    <img
      alt="evm_wallet.token_logo_alt"
      aria-hidden="true"
      class="c-wallet-balance-row__logo"
      height="40"
      src="https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png"
      width="40"
    />
    <div>
      <p
        class="q-mb-xs"
      >
        TLOS
      </p>
       Telos
    </div>
  </div>
  <div
    class="c-wallet-balance-row__right-container"
  >
    <div
      class="c-wallet-balance-row__balance-container"
    >
      <div
        class="c-wallet-balance-row__primary-amount"
      >
        $0.00 
        
        <inline-svg-stub
          aria-hidden="true"
          class="c-wallet-balance-row__info-icon"
          keepduringloading="true"
          src=""
          transformsource="[Function]"
        />
        <q-tooltip-stub
          model-value="false"
        >
          
          <span
            class="c-wallet-balance-row__tooltip"
          >
            evm_wallet.token_balance:
3,642.024300000000000000
          </span>
          
        </q-tooltip-stub>
        
      </div>
      <span
        class="c-wallet-balance-row__secondary-amount"
      >
        3.64K TLOS
      </span>
    </div>
    <q-btn-stub
      aria-label="evm_wallet.balance_row_actions_aria"
      class="c-wallet-balance-row__overflow"
      dense=""
      flat=""
      icon="more_vert"
      no-icon-animation=""
    >
      
      <q-menu-stub
        anchor="bottom end"
        offset="0,16"
        self="top right"
      >
        
        <ul
          class="c-wallet-balance-row__overflow-ul"
        >
          
          <li
            aria-labelledby="overflow-text-0"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon c-wallet-balance-row__overflow-icon--stroke"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-0"
            >
              evm_wallet.stake
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-1"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-1"
            >
              evm_wallet.buy
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-2"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-2"
            >
              evm_wallet.wrap
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-3"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-3"
            >
              evm_wallet.send
            </span>
          </li>
          
        </ul>
        
      </q-menu-stub>
      
    </q-btn-stub>
  </div>
</div>
`;

exports[`WalletBalanceRow.vue should render the right overflow items when the token is WTLOS 1`] = `
<div
  class="c-wallet-balance-row"
>
  <div
    class="c-wallet-balance-row__left-container"
  >
    <img
      alt="evm_wallet.token_logo_alt"
      aria-hidden="true"
      class="c-wallet-balance-row__logo"
      height="40"
      src="https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png"
      width="40"
    />
    <div>
      <p
        class="q-mb-xs"
      >
        WTLOS
      </p>
       Wrapped TLOS
    </div>
  </div>
  <div
    class="c-wallet-balance-row__right-container"
  >
    <div
      class="c-wallet-balance-row__balance-container"
    >
      <div
        class="c-wallet-balance-row__primary-amount"
      >
        $12.35K 
        
        <inline-svg-stub
          aria-hidden="true"
          class="c-wallet-balance-row__info-icon"
          keepduringloading="true"
          src=""
          transformsource="[Function]"
        />
        <q-tooltip-stub
          model-value="false"
        >
          
          <span
            class="c-wallet-balance-row__tooltip"
          >
            evm_wallet.fiat_value:
$12,345.67

evm_wallet.token_balance:
3,642.024300000000000000
          </span>
          
        </q-tooltip-stub>
        
      </div>
      <span
        class="c-wallet-balance-row__secondary-amount"
      >
        3.64K WTLOS
      </span>
    </div>
    <q-btn-stub
      aria-label="evm_wallet.balance_row_actions_aria"
      class="c-wallet-balance-row__overflow"
      dense=""
      flat=""
      icon="more_vert"
      no-icon-animation=""
    >
      
      <q-menu-stub
        anchor="bottom end"
        offset="0,16"
        self="top right"
      >
        
        <ul
          class="c-wallet-balance-row__overflow-ul"
        >
          
          <li
            aria-labelledby="overflow-text-0"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-0"
            >
              evm_wallet.unwrap
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-1"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-1"
            >
              global.contract
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-2"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-2"
            >
              evm_wallet.send
            </span>
          </li>
          
        </ul>
        
      </q-menu-stub>
      
    </q-btn-stub>
  </div>
</div>
`;

exports[`WalletBalanceRow.vue should render the right overflow items when the token is not a system token 1`] = `
<div
  class="c-wallet-balance-row"
>
  <div
    class="c-wallet-balance-row__left-container"
  >
    <img
      alt="evm_wallet.token_logo_alt"
      aria-hidden="true"
      class="c-wallet-balance-row__logo"
      height="40"
      src="https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png"
      width="40"
    />
    <div>
      <p
        class="q-mb-xs"
      >
        SHIB
      </p>
       Shiba
    </div>
  </div>
  <div
    class="c-wallet-balance-row__right-container"
  >
    <div
      class="c-wallet-balance-row__balance-container"
    >
      <div
        class="c-wallet-balance-row__primary-amount"
      >
        3.64K 
        
        <inline-svg-stub
          aria-hidden="true"
          class="c-wallet-balance-row__info-icon"
          keepduringloading="true"
          src=""
          transformsource="[Function]"
        />
        <q-tooltip-stub
          model-value="false"
        >
          
          <span
            class="c-wallet-balance-row__tooltip"
          >
            evm_wallet.token_balance:
3,642.024300000000000000

evm_wallet.no_fiat_value
          </span>
          
        </q-tooltip-stub>
        
      </div>
      <!--v-if-->
    </div>
    <q-btn-stub
      aria-label="evm_wallet.balance_row_actions_aria"
      class="c-wallet-balance-row__overflow"
      dense=""
      flat=""
      icon="more_vert"
      no-icon-animation=""
    >
      
      <q-menu-stub
        anchor="bottom end"
        offset="0,16"
        self="top right"
      >
        
        <ul
          class="c-wallet-balance-row__overflow-ul"
        >
          
          <li
            aria-labelledby="overflow-text-0"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-0"
            >
              global.contract
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-1"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-1"
            >
              evm_wallet.send
            </span>
          </li>
          
        </ul>
        
      </q-menu-stub>
      
    </q-btn-stub>
  </div>
</div>
`;
